### ISSUE TITLE
weak performance of TextBuilder insertion compare to StringBuilder

### ISSUE THREAD

**Issue by swrdlgc** [javolution/javolution#25]
I compare the insertion of TextBuilder and StringBuilder.
Result shows TextBuilder is weaker than StringBuilder in this case.
I think the reason is TextBuilder insertion calls setCharAt so many times(OriginalLength-InsertIndex+InsertLength), this is a huge of memory copy, and the deletion likewise. 
I suggest copy array reference, and each array may not fix length.

<pre>
/*
result:
TextBuilderInsertTime:   9289ms
StringBuilderInsertTime: 852ms
*/
private static final int N = 10000000;
	private static final int M = 50;
	
	public static void main(String[] args) {
		testTextBuilder();
		testStringBuilder();
	}

	static void testTextBuilder() {
		TextBuilder tb = new TextBuilder();
		for(int i = 0; i < N; ++i) {
			tb.append(i).append(',');
		}
		long c = System.currentTimeMillis();
		for(int i = 0; i < M; ++i) {
			tb.insert(0, "a");
		}
		System.err.println("TextBuilderInsertTime:   " + (System.currentTimeMillis() - c) + "ms");
	}
	
	static void testStringBuilder() {
		StringBuilder sb = new StringBuilder();
		for(int i = 0; i < N; ++i) {
			sb.append(i).append(',');
		}
		long c = System.currentTimeMillis();
		for(int i = 0; i < M; ++i) {
			sb.insert(0, "a");
		}
		System.err.println("StringBuilderInsertTime: " + (System.currentTimeMillis() - c) + "ms");
	}
</pre>

**Comment by dautelle** [comment]
Yes you are completely right. In the next version (7.0) TextBuilder will be based on FractalArray. It will be interesting to have your benchmark results run again as we should see significant performance improvements.
Note: Version 7.0 is almost ready, but still a few bugs to fix...
